src/dbally/views/pandas_base.py

```diff
@@ -19,48 +21,59 @@ class DataFrameBaseView(MethodsBaseView):
 
     def __init__(self, df: pd.DataFrame) -> None:
         """
+        Creates a new instance of the DataFrame view.
+
         Args:
-            df: Pandas DataFrame with the data to be filtered
+            df: Pandas DataFrame with the data to be filtered.
         """
         super().__init__()
         self.df = df
-
-        # The mask to be applied to the dataframe to filter the data
         self._filter_mask: Optional[pd.Series] = None
+        self._groupbys: Optional[Union[str, List[str]]] = None
+        self._aggregations: Optional[List[Tuple[str, str]]] = None
 
-    async def apply_filters(self, filters: IQLQuery) -> None:
+    async def apply_filters(self, filters: IQLFiltersQuery) -> None:
         """
         Applies the chosen filters to the view.
 
         Args:
-            filters: IQLQuery object representing the filters to apply
+            filters: IQLQuery object representing the filters to apply.
         """
-        self._filter_mask = await self.build_filter_node(filters.root)
+        self._filter_mask = await self._build_filter_node(filters.root)
 
-    async def build_filter_node(self, node: syntax.Node) -> pd.Series:
+    async def apply_aggregation(self, aggregation: IQLAggregationQuery) -> None:
+        """
+        Applies the aggregation of choice to the view.
+
+        Args:
+            aggregation: IQLQuery object representing the aggregation to apply.
+        """
+        self._groupbys, self._aggregations = await self.call_aggregation_method(aggregation.root)
```

Comment:
Looking at how those aggregation methods look like, for example:

```
    @view_aggregation()
    def mean_age_by_city(self) -> Tuple[str, List[Tuple[str, str]]]:
        return "city", [("age", "mean")]
```

It's very hard to tell looking at such code what "city" is, what "age" is, and what "mean" is. Maybe the return type should some sort of a special aggregation data class / named tuple? So that the individual elements are defined with names, showing what they are.

Reply:
Good idea, will implement this.